### PR TITLE
Consolidate organization settings

### LINKED ISSUE #2317: Consolidate organization settings into one place
# Goal

* All organization settings can be found from one place: the settings tab. This includes the settings in the header area
* Thee settings are organized in a meaningful way.
* There is a panel / group for (organization page) branding configuration panel in the organization settings.

# Definition of done
* [ ] All organization settings can be managed via Organization Settings tab 
* [ ] Organization Settings tab is laid out new way according the wireframe
* [ ] Organization settings are stored into database
* [ ] No Organization Edit button exists any more
* [ ] No possibility to modify Organization settings via modal any more
 

# Wireframe
organizationsettings4all

Items in the Setting tabs are organized as the following sections:
* **General**
- Organization Name
- Organization URL
- Description
- Logo Uploader
- Cover photo Uploader
* **Social Media Platform**
- Facebook
- Twitter
- Instagram
- LinkedIn
* **Contact Information**
- Contact Person's Name
- Phone
- Email
* **Content Management**
- No. of APIs to appear in default view
- No. of media Content to appear in default view
* **Adding Organization Managers**
* **Deleting Organization**

The bootstrap panels are removed from the design the keep a simple and clean view.
Instead, Header texts (h3) can be used to distinguish between sections with whitespace separating one section from another.
Separate sections are a preferable choice if in future new items needs to be added in settings tab.
Except for Organization Managers and Delete Organization sections, Save and Cancel buttons appears at each section.
[**Open Question:** Would it be possible to add  Global Save and Cancel buttons for General, Media, Content Management and Contact Information sections?]

### PR DESCRIPTION
**Pull request by matleppa**
Closes #2317

### CODE REVIEW

**Comment by nnn** [comment]
Please, avoid copy/paste code from other work-in-progress branches.
The problem here is that my code for cover image is not yet reviewed. 
It will created unnecessary complexity for the reviewer.

File: organizations/client/managers/form/form.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
 https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 -->
 
 <template name="addOrganizationManagerForm">
-  <h3>
+  <h3 class="orgSettingSubTitle">
```
Comment (marla-singer) [inline organizations/client/managers/form/form.html]: I think we don't need to change style for `<h3>` heading.
Remove class `orgSettingSubTitle`

File: organizations/client/profile/settings/contact_information/form.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  </h3>
+
+  {{# autoForm
+      collection=organizationCollection
+      doc=organization
+      id="orgSettingsContact"
+      type="update"
+  }}
+
+  {{> afQuickField name='contact.person' }}
```
Comment (marla-singer) [inline organizations/client/profile/settings/contact_information/form.html]: Make sure `autoForm` has two spaces for each level of nesting.

File: organizations/client/profile/settings/contact_information/form.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  {{> afQuickField name='contact.phone' }}
+  {{> afQuickField name='contact.email' }}
+
+  <button type="submit" class="btn btn-success" id="save-org-contact-settings">
+    {{_ "orgSettingsContactInfo_saveButton_text" }}
+  </button>
+  {{/ autoForm }}
+
+
+</template>
```
Comment (marla-singer) [inline organizations/client/profile/settings/contact_information/form.html]: Remove needless empty lines

File: organizations/client/profile/settings/contact_information/form.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<!-- Copyright 2017 Apinf Oy
+This file is covered by the EUPL license.
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 -->
+
+<template name="orgSettingsContactInfo">
```
Comment (marla-singer) [inline organizations/client/profile/settings/contact_information/form.html]: Set the full name as `organizationSettingsContactInfo`

File: organizations/client/profile/settings/general/form.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<!-- Copyright 2017 Apinf Oy
+This file is covered by the EUPL license.
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 -->
+
+<template name="orgSettingsGeneral">
```
Comment (marla-singer) [inline organizations/client/profile/settings/general/form.html]: Set the full name as `organizationSettingsGeneral`

File: organizations/client/profile/settings/general/form.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  </h3>
+
+  {{# autoForm
+      collection=organizationCollection
+      doc=organization
+      id="orgSettingsGeneral"
+      type="update"
+  }}
+
+  {{> afQuickField name='name' }}
```
Comment (marla-singer) [inline organizations/client/profile/settings/general/form.html]: Make sure `autoForm` has two spaces for each level of nesting.

File: organizations/client/profile/settings/general/form.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  {{> afQuickField name='url' }}
+  {{> afQuickField name='description' }}
+
+  <button type="submit" class="btn btn-success" id="save-org-general-settings">
+    {{_ "orgSettingsGeneral_saveButton_text" }}
+  </button>
+  {{/ autoForm }}
+
+</template>
```
Comment (marla-singer) [inline organizations/client/profile/settings/general/form.html]: Remove needless empty line

File: organizations/client/profile/settings/some_platforms/form.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<!-- Copyright 2017 Apinf Oy
+This file is covered by the EUPL license.
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 -->
+
+<template name="orgSettingsSomePlatforms">
```
Comment (marla-singer) [inline organizations/client/profile/settings/some_platforms/form.html]: Set the full name as `organizationSettingsSocialMedia`

File: organizations/client/profile/settings/some_platforms/form.html [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  </h3>
+
+  {{# autoForm
+      collection=organizationCollection
+      doc=organization
+      id="orgSettingsSomePlatforms"
+      type="update"
+  }}
+
+  {{> afQuickField name='socialMedia.facebook' }}
```
Comment (marla-singer) [inline organizations/client/profile/settings/some_platforms/form.html]: Make sure `autoForm` has two spaces for each level of nesting.

File: organizations/client/managers/form/form.html (hunk 1 above)
Comment (matleppa) [inline organizations/client/managers/form/form.html]: That style is common for every sections' subtitles in Organization settings page 
According to wireframe, now when the panels are not used any more, there were needed more space to before and after subtitles so they act as delimiters between sections.

Comment (marla-singer) [inline organizations/client/managers/form/form.html]: > there were needed more space to before and after subtitles so they act as delimiters between sections.

For this you have to use `<div>` tag but not add `margin` property for heading

**Comment by marla-singer** [comment]
@matleppa You should contain each part (General, Logo, Social and etc) in `<div>` block and this `<div>` block will have `margin-bottom` property for separating

File: organizations/client/profile/settings/some_platforms/form.html [hunk 10]
```diff
@@ -9,10 +9,10 @@ <h3 class="organizationSettingSubTitle">
...
-      doc=organization
-      id="organizationSettingsSomePlatforms"
-      type="update"
+    collection=organizationCollection
+    doc=organization
+    id="organizationSettingsSomePlatforms"
+    type="update"
   }}
 
   {{> afQuickField name='socialMedia.facebook' }}
```
Comment (marla-singer) [inline organizations/client/profile/settings/some_platforms/form.html]: I meant `{{> afQuickField name='socialMedia.facebook' }}` fields, not properties.

Must be
```html
  {{# autoForm
    collection=organizationCollection
    doc=organization
    id="organizationSettingsSomePlatforms"
    type="update"
  }}
    
    {{> afQuickField name='socialMedia.facebook' }}
    {{> afQuickField name='socialMedia.twitter' }}
    {{> afQuickField name='socialMedia.instagram' }}
    {{> afQuickField name='socialMedia.linkedIn' }}
    
    <button type="submit" class="btn btn-success" id="save-org-some-settings">
      {{_ "organizationSettingsSomePlatforms_saveButton_text" }}
    </button>
  {{/ autoForm }}
```

File: organizations/client/profile/settings/some_platforms/form.html [hunk 11]
```diff
@@ -0,0 +1,27 @@
+<!-- Copyright 2017 Apinf Oy
+This file is covered by the EUPL license.
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 -->
+
+<template name="organizationSettingsSomePlatforms">
```
Comment (marla-singer) [inline organizations/client/profile/settings/some_platforms/form.html]: Set the full name as organizationSettingsSocialMedia

File: organizations/client/profile/settings/some_platforms/form.html [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  <h3 class="organizationSettingSubTitle">
```
Comment (marla-singer) [inline organizations/client/profile/settings/some_platforms/form.html]: Remove class `organizationSettingSubTitle`.

File: organizations/client/profile/profile.css [hunk 13]
```diff
@@ -12,7 +12,6 @@ https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence
   margin-bottom: 1.5em;
 }
 
-.organizationSettingSubTitle {
-  margin-top: 1.6em;
-  margin-bottom: 1.6em;
+.organizationSettingSection {
```
Comment (marla-singer) [inline organizations/client/profile/profile.css]: Don't use camelCase for class name or id name.
Rename class as `setting-section`

Comment (marla-singer) [inline organizations/client/profile/profile.css]: Also add rule as
```css
.setting-section form {
  margin-top: 1.6em;
}
```
It provides margin between form and heading without additional class

File: organizations/client/profile/settings/settings.html [hunk 14]
```diff
@@ -10,20 +10,22 @@
...
+          </p>
+          <p>
+            {{_ "organizationSettings_uploadLogoText" }}
+          </p>
+
+          {{> uploadOrganizationLogo organization=organization }}
+        </div>
 
         <!-- <b>
           {{_ "organizationSettings_uploadCover_heading" }}
```
Comment (marla-singer) [inline organizations/client/profile/settings/settings.html]: Clean all code which relates to Cover Photo

File: organizations/client/profile/settings/settings.html [hunk 15]
```diff
@@ -37,8 +39,12 @@ <h3 class="organizationSettingSubTitle">
         {{> organizationSettingsContactInfo organization=organization }}
 
         <!-- {{> organizationSettingsContentManagement organization=organization }} -->
```
Comment (marla-singer) [inline organizations/client/profile/settings/settings.html]: Clean all code which relates to Cover Photo

File: organizations/client/profile/header/header.css [hunk 16]
```diff
@@ -0,0 +1,18 @@
+.organization-cover-block {
+  position: absolute;
```
Comment (marla-singer) [inline organizations/client/profile/header/header.css]: Clean all code which relates to Cover Photo

File: organizations/client/profile/header/header.html [hunk 17]
```diff
@@ -5,21 +5,23 @@
 
 <template name="organizationProfileHeader">
   <div class="page-head-topblock">
+    <!-- <div class="organization-cover-block">
+      {{> viewOrganizationCover organization=organization }}
+    </div> -->
```
Comment (marla-singer) [inline organizations/client/profile/header/header.html]: Clean all code which relates to Cover Photo

File: organizations/client/profile/header/header.html [hunk 18]
```diff
@@ -5,21 +5,23 @@
...
     <div class="container">
       <div class="row">
         <div id="organization-file-browse" class="col-md-2 profile-img-container">
           {{> viewOrganizationLogo organization=organization }}
           <span class="fa fa-camera"></span>
         </div>
+        <!-- <button id="organization-cover-file-browse" class="col-md-2 btn btn-default pull-right">
+          <span class="fa fa-camera"></span>
+          {{_ "organizationProfileHeader_uploadCoverButton_text" }}
+        </button> -->
```
Comment (marla-singer) [inline organizations/client/profile/header/header.html]: Clean all code which relates to Cover Photo

File: organizations/client/profile/header/header.js [hunk 19]
```diff
@@ -11,10 +11,14 @@ import { Modal } from 'meteor/peppelg:bootstrap-3-modal';
...
 import OrganizationLogo from '/organizations/logo/collection/collection';
+// import OrganizationCover from '/organizations/cover/collection/collection';
 
 Template.organizationProfileHeader.onRendered(function () {
   // Assign resumable browse to element
   OrganizationLogo.resumable.assignBrowse(this.$('#organization-file-browse'));
+
+  // Assign resumable browse to element
+  // OrganizationCover.resumable.assignBrowse(this.$('#organization-cover-file-browse'));
 });
```
Comment (marla-singer) [inline organizations/client/profile/header/header.js]: Clean all code which relates to Cover Photo

**Comment by marla-singer** [comment]
@matleppa Resolve conflict in **your branch**

File: core/lib/i18n/en.i18n.json [hunk 20]
```diff
@@ -450,14 +450,29 @@
...
-  "organizationSettingsDelete_text_information": "The organization information will be removed and this action cannot be undone.",
-  "organizationSettingsDelete_title": "Delete the Organization profile",
+  "organizationSettingsDelete_text_information": "The organization information will be removed and all APIs will be disconnected. This action cannot be undone.",
+  "organizationSettingsDelete_title": "Delete Organization",
   "organizationSettingsDelete_panelTitle": "Delete",
+  "organizationSettingsContactInfo_cancelButton_text": "Cancel",
+  "organizationSettingsContactInfo_saveButton_text": "Save",
+  "organizationSettingsContactInfo_title": "Contact Information",
+  "organizationSettingsContentManagement_cancelButton_text": "Cancel",
+  "organizationSettingsContentManagement_numberOfApisPerPage_description": "Number of APIs per page",
```
Comment (marla-singer) [inline core/lib/i18n/en.i18n.json]: @matleppa Why did you add i18n tags for Content Managment block if you even doesn't have this block in Settings page? 
Remove all tags related to Content Management block

File: organizations/client/profile/settings/settings.html [hunk 21]
```diff
@@ -7,42 +7,45 @@
...
+
+        {{> organizationSettingsGeneral organization=organization }}
+
+        <div class="setting-section">
+          <h3>
+            {{_ "organizationSettingsCoverAndLogo_title" }}
+          </h3>
+
+          <p>
+            <b>
```
Comment (marla-singer) [inline organizations/client/profile/settings/settings.html]: Instead of `<p><b></b></p>` use `<label></labe>` tag

File: organizations/client/profile/settings/settings.html [hunk 22]
```diff
@@ -7,42 +7,45 @@
...
-          </div>
+            </b>
+          </p>
+          <p>
+            {{_ "organizationSettings_uploadLogoText" }}
+          </p>
+
+          {{> uploadOrganizationLogo organization=organization }}
+
+          <h3>
```
Comment (marla-singer) [inline organizations/client/profile/settings/settings.html]: Change `<h3>` tag to `<label>` tag and change text to "Cover photo"
